build.gradle

```diff
@@ -258,6 +262,39 @@ subprojects {
   }
 
   jar {
+
+    doFirst {
+      //version embedding at build time
+      //we create a file called __Versioning__[md5] and set its payload to contain versioning info
+      //
+      //the file name is not fixed, but contains an md5 hash of the project group name and version.
+      //its not fixed so that if multiple libraries are repackaged (think fat/shaded jar) the versioning files
+      //are unlikely to conflict or get overwritten. the md5 hash part is chosen such that builds are
+      //"repeatable" - re-running the same build results in the same output. this is nice for incremental builds.
+      //
+      //the value contains versioning information for both this module and the entire project it was built as part of.
+      //
+      //this information allows us, at runtime, to determine what libraries/versions exist on a given classpath
+
+      MessageDigest md = MessageDigest.getInstance("MD5")
+      md.update(String.valueOf(project.group).getBytes("UTF-8"))
+      md.update(String.valueOf(project.name).getBytes("UTF-8"))
```

Comment:
Why do we need hash of (group, name, version)?  maybe we can use _li_versioning_{VersionNumber}?  LI prefix to avoid conflict with upstream.  This makes filename readable and easy for scripting?

Reply:
group + name + bversion is required to avoid conflict with other projects (so group), other jars in the same project (so project name) and other versions of the same jar on the classpath at the same time (so version).

our group and versions are different to upstream, so i dont expect conflicts.
also, upstream doesnt container this data (and im not currently planning on opening a PR for this upstream)

Reply:
Ok. hashing makes it hard to read.  I am fine either way. hashing or li_versionning_group_name_bversion